packages/truffle/test/scenarios/commandRunner.js

```diff
@@ -74,17 +79,20 @@ module.exports = {
 
       child.stdout.on("data", data => {
         // accumulate buffer from chunks
-        if (!seenChildPrompt) {
-          outputBuffer += data;
-        }
+        outputBuffer += data;
 
-        // child process is ready for input when it displays the readyPrompt
-        if (!seenChildPrompt && outputBuffer.includes(readyPrompt)) {
-          seenChildPrompt = true;
-          inputCommands.forEach(command => {
-            child.stdin.write(command + EOL);
-          });
-          child.stdin.end();
+        // count prompt
+        const foundPrompts = (outputBuffer.match(readyPromptRex) || []).length;
+        if (foundPrompts > numSeenPrompts) {
+          numSeenPrompts++;
```

Comment:
This should probably be:

```suggestion
          numSeenPrompts = foundPrompts;
```

Reply:
The most correct! Thanks, @davidmurdoch.